runtime/src/bank.rs

```diff
@@ -4299,30 +4299,37 @@ impl Bank {
         let mut time_collecting_rent_us = 0;
         let mut time_storing_accounts_us = 0;
         let can_skip_rewrites = self.bank_hash_skips_rent_rewrites();
-        let test_skip_rewrites_but_include_hash_in_bank_hash = !can_skip_rewrites
-            && self
-                .rc
-                .accounts
-                .accounts_db
-                .test_skip_rewrites_but_include_in_bank_hash;
+        let test_skip_rewrites_but_include_in_bank_hash = self
+            .rc
+            .accounts
+            .accounts_db
+            .test_skip_rewrites_but_include_in_bank_hash;
         let mut skipped_rewrites = Vec::default();
         for (pubkey, account, _loaded_slot) in accounts.iter_mut() {
+            let rent_epoch_pre = account.rent_epoch();
             let (rent_collected_info, collect_rent_us) = measure_us!(collect_rent_from_account(
                 &self.feature_set,
                 &self.rent_collector,
                 pubkey,
                 account
             ));
             time_collecting_rent_us += collect_rent_us;
+            let rent_epoch_post = account.rent_epoch();
+
+            // did the account change in any way due to rent collection?
+            let account_changed =
+                rent_collected_info.rent_amount != 0 || rent_epoch_post != rent_epoch_pre;
```

Comment:
nit: how about report a metric for 'account_changed', so that we can monitor it when this happens?

Reply:
Interesting, do you know the name of the metric? 
maybe we can check that metric in the log of pop256 from your last failed run?

Reply:
I see. Thanks for checking this.
fwik, rent_collected_info.rent_amount is already covered. 
maybe we just need a new counter for rent_epoch_post != rent_epoch_pre?

Reply:
I think it will indicates that we have a non-rent exempt account somehow. 
In theory, every accounts in mnb now should have rent_epoch set to max. 
But when we add new special accounts in future, i.e. new sysvar, there could be a chance that we construct the account with non-max rent_epoch? If this happens, the new metrics will tell us.